opencti-platform/opencti-front/src/private/components/settings/attributes/VocabularyCreation.tsx

```diff
@@ -0,0 +1,214 @@
+import Fab from '@mui/material/Fab';
+import { Add, Close } from '@mui/icons-material';
+import React, { FunctionComponent, useState } from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { Field, Form, Formik, FormikConfig } from 'formik';
+import Button from '@mui/material/Button';
+import Drawer from '@mui/material/Drawer';
+import { graphql, useMutation } from 'react-relay';
+import * as Yup from 'yup';
+import TextField from '../../../../components/TextField';
+import type { Theme } from '../../../../components/Theme';
+import { useFormatter } from '../../../../components/i18n';
+import { VocabularyCreationMutation } from './__generated__/VocabularyCreationMutation.graphql';
+import { insertNode } from '../../../../utils/store';
+import { VocabulariesLines_DataQuery$variables } from './__generated__/VocabulariesLines_DataQuery.graphql';
+import useVocabularyCategory from '../../../../utils/hooks/useVocabularyCategory';
+import AutocompleteField from '../../../../components/AutocompleteField';
+import { fieldSpacingContainerStyle } from '../../../../utils/field';
+import { Option } from '../../common/form/ReferenceField';
+
+interface VocabularyCreationProps {
+  paginationOptions: VocabulariesLines_DataQuery$variables,
+}
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  drawerPaper: {
+    minHeight: '100vh',
+    width: '50%',
+    position: 'fixed',
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    padding: 0,
+  },
+  createButton: {
+    position: 'fixed',
+    bottom: 30,
+    right: 230,
+  },
+  closeButton: {
+    position: 'absolute',
+    top: 12,
+    left: 5,
+    color: 'inherit',
+  },
+  header: {
+    backgroundColor: theme.palette.background.nav,
+    padding: '20px 20px 20px 60px',
+  },
+  container: {
+    padding: '10px 20px 20px 20px',
+  },
+  buttons: {
+    marginTop: 20,
+    textAlign: 'right',
+  },
+  button: {
+    marginLeft: theme.spacing(2),
+  },
+}));
+
+const vocabularyAdd = graphql`
+  mutation VocabularyCreationMutation($input: VocabularyAddInput!) {
+    vocabularyAdd(input: $input) {
+      ...useVocabularyCategory_Vocabularynode
```

Comment:
Yep, because the vocabularyDefinitions is static, I added a hook to prevent to call them everytime we want it but only once. Therefore some of my Queries are in a hook